FirebaseInstallations/Source/Tests/FIRInstallationsTests.m

```diff
@@ -0,0 +1,138 @@
+
+- (void)setUp {
+  [super setUp];
+
+  self.installations = [[FIRInstallations alloc] initWithGoogleAppID:@"GoogleAppID"
+                                                             appName:@"appName"];
+}
+
+- (void)tearDown {
+  self.installations = nil;
+  [super tearDown];
+}
+
+- (void)testInstallationsWithApp {
+  [self assertInstallationsWithAppNamed:@"testInstallationsWithApp1"];
+  [self assertInstallationsWithAppNamed:@"testInstallationsWithApp2"];
+
+  [FIRApp resetApps];
+}
+
+- (void)testInstallationIDSuccess {
+  XCTestExpectation *idExpectation = [self expectationWithDescription:@"InstallationIDSuccess"];
+  [self.installations
+      installationIDWithCompletion:^(NSString *_Nullable identifier, NSError *_Nullable error) {
+        XCTAssertNil(error);
+        XCTAssertNotNil(identifier);
+        XCTAssertGreaterThan(identifier.length, 0);
+
+        [idExpectation fulfill];
+      }];
+
+  [self waitForExpectations:@[ idExpectation ] timeout:0.5];
+}
+
+- (void)testAuthTokenSuccess {
+  XCTestExpectation *tokenExpectation = [self expectationWithDescription:@"AuthTokenSuccess"];
+  [self.installations
+      authTokenWithCompletion:^(FIRInstallationsAuthTokenResult *_Nullable tokenResult,
+                                NSError *_Nullable error) {
+        XCTAssertNotNil(tokenResult);
+        XCTAssertGreaterThan(tokenResult.authToken.length, 0);
+        XCTAssertTrue([tokenResult.expirationTime laterDate:[NSDate date]]);
+        XCTAssertNil(error);
+
+        [tokenExpectation fulfill];
+      }];
+
+  [self waitForExpectations:@[ tokenExpectation ] timeout:0.5];
+}
+
+- (void)testAuthTokenForcingRefreshSuccess {
+  XCTestExpectation *tokenExpectation = [self expectationWithDescription:@"AuthTokenSuccess"];
+  [self.installations
+      authTokenForcingRefresh:YES
+                   completion:^(FIRInstallationsAuthTokenResult *_Nullable tokenResult,
+                                NSError *_Nullable error) {
+                     XCTAssertNotNil(tokenResult);
+                     XCTAssertGreaterThan(tokenResult.authToken.length, 0);
+                     XCTAssertTrue([tokenResult.expirationTime laterDate:[NSDate date]]);
+                     XCTAssertNil(error);
+
+                     [tokenExpectation fulfill];
+                   }];
+
+  [self waitForExpectations:@[ tokenExpectation ] timeout:0.5];
+}
+
+- (void)testDeleteSuccess {
+  XCTestExpectation *deleteExpectation = [self expectationWithDescription:@"DeleteSuccess"];
+  [self.installations deleteWithCompletion:^(NSError *_Nullable error) {
+    XCTAssertNil(error);
+    [deleteExpectation fulfill];
+  }];
+
+  [self waitForExpectations:@[ deleteExpectation ] timeout:0.5];
+}
+
+#pragma mark - Common
+
+- (FIRInstallations *)assertInstallationsWithAppNamed:(NSString *)appName {
+  FIRApp *app = [self createAndConfigureAppWithName:appName];
+  FIRInstallations *installations = [FIRInstallations installationsWithApp:app];
+
```

Comment:
Nit: `XCTAssertNotNil(installations);` just in case the container doesn't work as expected.